tests/unified/runner.py

```
@@ -66,6 +68,102 @@ class TestExecutionError(Exception):
+
+        # AWS credentials are configured via OIDC in GitHub Actions
+        # Check if boto3 can access credentials (either from environment or OIDC)
+        try:
+            import boto3
+
+            session = boto3.Session()
+            credentials = session.get_credentials()  # pyright: ignore
+            if not credentials:
+                logger.warning("No AWS credentials available, skipping S3 upload")
+                return
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)